### ISSUE TITLE
Provide "Motions" with unique names

### ISSUE THREAD

**Issue by tkuchida** [opensim-org/opensim-gui#609]
### Steps to reproduce
- start OpenSim
- open a model
- run click :running_man: button
- run Forward Tool
- run Analyze Tool
- run Static Tool
- run Static Tool

### Expected result
Navigator shows one entry for each operation and each item in the list is unique, such as
- Results_Forward_001
- Results_Forward_002
- Results_Analyze_001
- Results_Static_001
- Results_Static_002

Could also use a time stamp.

### Actual result
Several items are identical and some are "Results" while others are "states":
- Results
- Results
- states
- states
- states

![](https://user-images.githubusercontent.com/4203505/37687084-6dc6919a-2c57-11e8-8a09-b1013f3e3208.png)

It would be helpful to provide unique names by default, and perhaps to hide the distinction between "Results" and "states" (unless the user is expected to know the distinction).

### Environment and GUI version
AppVeyor artifact OpenSim-0d41eef3-2018-03-14.zip

**Comment by aymanhab** [comment]
@tkuchida Good idea, some issues involved:
- We probably want the name displayed to be different from what's in the file otherwise we're changing user files.
- Would a time stamp in some form e.g. yymmddhhmmss be an option?
- Would you like to encode how the file was generated (as suggested above)? If so this can be easily done by the GUI for results created in it but not for files managed by the API directly.

**Comment by tkuchida** [comment]
The GUI doesn't have enough information to automatically assign a maximally meaningful name. For example, if a user is running a series of Forward simulations at different tendon slack lengths, the GUI doesn't know that the Results should be labeled with the tendon slack length. It is not possible to avoid requiring the user to edit the name for it to be truly meaningful. The current issues are that several items have identical names and that the user is exposed to a low-level detail (Results vs. states) that has no meaning to them. An issue occurs in the following use case:
- open a model
- run a few Forward simulations, playing the animation after each one but not paying attention to the Navigator
- open the plotter to compare results
- discover several items named "Results" in the drop-down list

At this point, the user has two choices. First is to trust that the GUI has populated the drop-down list in the same order as the Results were generated. There is no reason to believe the GUI does that—in fact, ordering the Results with the most recent at the top would be equally plausible. Second option is to rename the Results in the Navigator (i.e. satisfy the needs of the GUI) before they can proceed with their work.

> Would a time stamp in some form e.g. yymmddhhmmss be an option?

May not be immediately obvious. I would use [something standard](https://en.wikipedia.org/wiki/ISO_8601) like `yyyy-mm-dd hh:mm:ss`.

**Comment by chrisdembia** [comment]
I like the completeness of the time stamp, but I think it's common for software to handle this situation by just incrementing a counter (i.e., `Results`, `Results 1`, `Results 2`, ...).

**Comment by tkuchida** [comment]
> I think it's common for software to handle this situation by just incrementing a counter

I agree and that would be my first choice, but if Ayman wants to use a date/time stamp, I recommend using a format that is immediately recognizable (and there are already widely accepted standard formats).

**Comment by aseth1** [comment]
:+1: @tkuchida - I really like your proposal for the expected behavior. Especially that results are associated/named according to the means in which they were produced. The distinction between a loaded motion and the generated resultant motion by running a Tool (or🏃) is also helpful. 

What should we do in the cases when the motion that appears in the Navigator is not a result at all. In the case of `AnalyzeTool` or `Inverse Dynamics` the results are the quantities computed (written to file) that are based on Tool inputs (motions and forces).  In this case calling the reproduced motion *Results_XXXX_###* (e.g. `Results_ID_001`) might be confusing - it confuses me. Why should a motion/states file even appear in the Navigator in that case? If the Tool computes state values like coordinates and activations of the model, it makes sense to have a synthesis of the results that are nicely visualized and played back.  But for Analyses (`AnalyzeTool`) and Inverse Dynamics, i'm not sure they should "produce" additional motions.

**Comment by tkuchida** [comment]
> What should we do in the cases when the motion that appears in the Navigator is not a result at all.

There are many options; I have described two below. I don't know the details of the GUI's design philosophy, so I don't know which option is closest to that vision.

:one: If a user wants to run an analysis on existing results, it seems like the current workflow would be to open the model, open the Analyze Tool, run the analysis, then play the animation that appears in the Navigator to see what was just analyzed. That's backwards: the user should (and should want to) load/inspect the motion first, then run the analysis on the loaded motion (either by selecting the corresponding item using the "Loaded motion" drop-down box, or by specifying the filename in the "From file" textbox). This workflow suggests that the Analyze Tool in the GUI should not take a file as input: it should require the user to load the motion first and use the "Loaded motion" drop-down box. The GUI would then be forcing the user to follow a more careful workflow, and would also avoid potentially loading a motion, playing the animation, then accidentally pointing the Tool to the wrong file.

:two: Organize Results/states/Motions in the Navigator. Rather than a generic "Motions" heading, there could be a separate heading for each Tool, perhaps something like
<pre>
...
+ Controllers
+ Probes
- Tool Outputs
    + Scale Tool
    + Inverse Kinematics Tool
...
</pre>
Each Tool could then produce artifacts in the Navigator without potentially confusing the user about which Results/states/Motions are newly generated, loaded from files, etc. One of the headings could even be "States loaded by the Analyze Tool".
